plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/SemiTransactionalHiveMetastore.java

```diff
@@ -590,20 +582,23 @@ public synchronized void dropTable(ConnectorSession session, String databaseName
         Action<TableAndMore> oldTableAction = tableActions.get(schemaTableName);
         if (oldTableAction == null || oldTableAction.getType() == ActionType.ALTER) {
             HdfsContext hdfsContext = new HdfsContext(session);
-            tableActions.put(schemaTableName, new Action<>(ActionType.DROP, null, hdfsContext, session.getQueryId()));
+            if (deleteData) {
+                tableActions.put(schemaTableName, new Action<>(ActionType.DROP, null, hdfsContext, session.getQueryId()));
+            }
+            else {
+                tableActions.put(schemaTableName, new Action<>(ActionType.DROP_PRESERVE_DATA, null, hdfsContext, session.getQueryId()));
+            }
```

Comment:
```suggestion
ActionType actionType = deleteData ? ActionType.DROP : ActionType.DROP_PRESERVE_DATA;
tableActions.put(schemaTableName, new Action<>(actionType, null, hdfsContext, session.getQueryId()));
```

Reply:
Also got that on my mind, however I decided otherwise. Maybe I was just following it (https://github.com/trinodb/trino/blob/master/.github/DEVELOPMENT.md#avoid-ternary-operator) too harshly.